File: HyprMX-Android-ExampleApp/app-java/src/main/java/com/hyprmx/android/exampleapp/HyprMXBannerListenerImpl.java
package com.hyprmx.android.exampleapp;

import android.util.Log;

import androidx.annotation.NonNull;

import com.hyprmx.android.sdk.banner.HyprMXBannerListener;
import com.hyprmx.android.sdk.banner.HyprMXBannerView;

public class HyprMXBannerListenerImpl implements HyprMXBannerListener {

  /**
   * Called when an user clicks on an ad for a given banner placement.
   */
  @Override
  public void onAdClicked(@NonNull HyprMXBannerView hyprMXBannerView) {
    Log.d(MainActivity.TAG, "onAdClicked for " + hyprMXBannerView.getPlacementName());
  }

  /**
   * Called when an ad is closed. Your application should resume here.
   */
  @Override
  public void onAdClosed(@NonNull HyprMXBannerView hyprMXBannerView) {
    Log.d(MainActivity.TAG, "onAdClosed for " + hyprMXBannerView.getPlacementName());
  }

  /**
   * Called when an user has navigated away from the app.
   */
  @Override
  public void onAdLeftApplication(@NonNull HyprMXBannerView hyprMXBannerView) {
    Log.d(MainActivity.TAG, "onAdLeftApplication for " + hyprMXBannerView.getPlacementName());
  }

  /**
   * Called when an user opens on an ad for a given banner placement.
   */
  @Override
  public void onAdOpened(@NonNull HyprMXBannerView hyprMXBannerView) {
    Log.d(MainActivity.TAG, "onAdOpened for " + hyprMXBannerView.getPlacementName());
  }

  /**
   * Called when an ad was impressed.
   */
  @Override
  public void onAdImpression(@NonNull HyprMXBannerView hyprMXBannerView) {
    Log.d(MainActivity.TAG, "onAdOpened for " + hyprMXBannerView.getPlacementName());
  }
}

File: HyprMX-Android-ExampleApp/app-kotlin/src/main/java/com/hyprmx/android/exampleapp/HyprMXBannerListenerImpl.kt
package com.hyprmx.android.exampleapp

import android.util.Log
import com.hyprmx.android.sdk.banner.HyprMXBannerListener
import com.hyprmx.android.sdk.banner.HyprMXBannerView

class HyprMXBannerListenerImpl : HyprMXBannerListener {
  /**
   * Called when an user clicks on an ad for a given banner placement.
   */
  override fun onAdClicked(view: HyprMXBannerView) {
    Log.d(MainActivity.TAG, "onAdClicked for ${view.placementName}")
  }

  /**
   * Called when an user opens on an ad for a given banner placement.
   */
  override fun onAdOpened(view: HyprMXBannerView) {
    Log.d(MainActivity.TAG, "onAdOpened for ${view.placementName}")
  }

  /**
   * Called when an ad is closed. Your application should resume here.
   */
  override fun onAdClosed(view: HyprMXBannerView) {
    Log.d(MainActivity.TAG, "onAdClosed for ${view.placementName}")
  }

  /**
   * Called when an user has navigated away from the app.
   */
  @Suppress("OVERRIDE_DEPRECATION")
  override fun onAdLeftApplication(view: HyprMXBannerView) {
    Log.d(MainActivity.TAG, "onAdLeftApplication for ${view.placementName}")
  }

  /**
   * Called when an ad was impressed.
   */
  override fun onAdImpression(view: HyprMXBannerView) {
    Log.d(MainActivity.TAG, "onAdImpression for ${view.placementName}")
  }
}
